README.md

```diff
@@ -26,28 +26,104 @@
+
+* API USE CASES
+    * Restricted API #1 - All On-heap
+        * wrapped heap arrays
+        * wrapped on-heap ByteBuffers 
+        * No off-heap ByteBuffers
+        * No allocations of off-heap memory
+        * No use of memory-mapped files
+    * Restricted API #2 - Add off-heap ByteBuffers
+        * wrapped heap arrays
+        * wrapped on-heap ByteBuffers 
+        * wrapped off-heap ByteBuffers
+        * No allocations of off-heap memory
+        * No use of memory-mapped files
+    * Full API
+        * wrapped heap arrays
+        * wrapped on-heap ByteBuffers 
+        * wrapped off-heap ByteBuffers
+        * allocations of off-heap memory
+        * use of memory-mapped files
+
+* CALLING APPLICATION JAVA VERSION & JPMS APPLICATION
+    * Running Java 8
+    * Running Java 9 - 13, non-JPMS application
+    * Running Java 9 - 13, JPMS application
+
+
+### Table of API Use Cases Compared to Application Java Version and JPMS Application, and Possible JVM Arguments Required
+
+| Java Version & JPMS      | Restricted API 1 | Restricted API 2 API | Full API |
+|:------------------------:|:----------------:|:--------------------:|:--------:|
+| Java 8                   |   Args 1         |  Args 1              | Args 1   |
+| Java 9-13, non-JPMS      |   Args 1         |  Args 1              | Args 3   |
+| Java 9-13, JPMS          |   Args 1         |  Args 2              | Args 4   |
+
+### Args 1
+No JVM arguments required
+
+### Args 2
+User must supply the following argument to the JVM:
+
+* --add-opens java.base/sun.nio.ch=org.apache.datasketches.memory
+
+### Args 3
+User must supply the following arguments to the JVM:
+
+* --add-exports java.base/jdk.internal.misc=ALL-UNNAMED
+* --add-exports java.base/jdk.internal.ref=ALL-UNNAMED
+* --add-opens java.base/java.nio=ALL-UNNAMED
+* --add-opens java.base/sun.nio.ch=ALL-UNNAMED
+
+### Args 4
+User must supply the following arguments to the JVM:
+
+* --add-exports java.base/jdk.internal.misc=org.apache.datasketches.memory
+* --add-exports java.base/jdk.internal.ref=org.apache.datasketches.memory
+* --add-opens java.base/java.nio=org.apache.datasketches.memory
+* --add-opens java.base/sun.nio.ch=org.apache.datasketches.memory
+
+
+## DEVELOPER USAGE
+In this environent the developer needs to build the Memory component from source and run the Memory Component tests.  There are two use cases. The first is for a *System Developer* that needs to build and test their own Jar from source for a specific Java version. The second use case is for a *Memory Component Developer and Contributor*. 
+
+* System Developer
+    * Compile, test and create a Jar for a specific Java version
+        * use the privided script for this purpose 
+
+* Memory Component Developer / Contributor
+    * Compile & test the library from source code using:
+        * Eclipse (version)
+        * IntelliJ (version)
+        * Maven (version)
+        * Command-line
```

Comment:
Does command-line here refer to the provided script?

Reply:
Yes it could, but I was thinking about my toy org.xyz example, we could say Command-line scripts ?